safe_transaction_service/history/migrations/0068_alter_multisigtransaction_origin.py

```diff
@@ -0,0 +1,48 @@
+# Generated by Django 4.1.2 on 2022-11-14 18:39
+
+import json
+
+from django.db import migrations, models
+
+
+def convert_str_to_json(apps, schema_editor):
+    MultisigTransaction = apps.get_model("history", "MultisigTransaction")
+    transactions = (
+        MultisigTransaction.objects.exclude(origin__icontains="{")
+        .filter(origin__isnull=False)
+        .iterator()
+    )
+    for transaction in transactions:
+        transaction.origin = json.dumps(transaction.origin)
+        transaction.save()
+
+
+def convert_jsonfield_to_charfield(apps, schema_editor):
+    MultisigTransaction = apps.get_model("history", "MultisigTransaction")
+    MultisigTransaction.objects.filter(origin__exact="{}").update(origin=None)
+    transactions = (
+        MultisigTransaction.objects.exclude(origin__icontains="{")
+        .filter(origin__isnull=False)
+        .iterator()
+    )
+    for transaction in transactions:
+        transaction.origin = json.loads(transaction.origin)
```

Comment:
Here you have an `object` and you try to call `json.loads` , this will fail

Reply:
AFAIK the reverse migration undo the migration executing the operations in a reverse way then we have not a object we have an string.

Reply:
The reason for this is because the django migration from `jsonfield` to `charfield` applies a json.loads for each row, then for example for "hello" is going to store in database '"hello"' instead  of "hello":
```
json.dumps("hello")
Out[34]: '"hello"'
```

Reply:
Ok, but then an object will be stored as an object instead of a stringfied json, right?

Reply:
This is the backward migration objects should be stored like `stringfield` json and strings should stored like strings.

Reply:
Ok